File: controllers/patient.py
from flask import Flask
from flask_restful import Resource, Api, request

# go to desired directory
import sys
sys.path.append("../")

# imports 
import db_repo.patient as pat



class Patient(Resource):
	
	def __init__(self):
		pass

	def get(self):
		patient_list = []
		patients = pat.view_patients()
		for patient in patients:
			patient_list.append(patient)

		return {
			"patients": patient_list
		}
		

	def post(self):
		if request.method == 'POST':
			# capturing from postman
			first_name = request.form.get('first_name')
			middle_name = request.form.get('middle_name')
			last_name = request.form.get('last_name')	
			ssn = request.form.get('ssn')
			next_of_kin = request.form.get('next_of_kin')
			home_phone = request.form.get('home_phone')
			work_phone = request.form.get('work_phone')
			symptoms = request.form.get("idk")
			health_insurance = request.form.get("health_insurance")
			vaccination = request.form.get("vaccination")
			vaccination_date = request.form.get("vaccination_date")
			drugs_alcohol = request.form.get("drugs_alchohol")
			sexually_active = request.form.get("sexually_active")
			allergies = request.form.get("allergies")
			blood_type = request.form.get("blood_type")
			notes = request.form.get("notes")
			date_of_birth = request.form.get("date_of_birth")
			height = request.form.get('height')
			weight = request.form.get('weight')
			

			# create patient in database
			dict_input = {
            	"ssn": ssn,
            	"first_name": first_name,
            	"middle_name": middle_name,
            	"last_name": last_name,
            	"date_of_birth": date_of_birth,
            	"height": height,
            	"weight": weight,
            	"next_of_kin": next_of_kin,
            	"home_phone": home_phone,
            	"work_phone": work_phone,
            	"symptoms": symptoms,
            	"health_insurance": health_insurance,
            	"vaccination": vaccination,
            	"vaccination_date": vaccination_date,
            	"drugs_alchohol": drugs_alcohol,
            	"sexually_active": sexually_active,
            	"allergies": allergies,
            	"blood_type": blood_type,
            	"notes": notes
        	}

			

			result = pat.create_patient(dict_input)

			# check if creating patient worked
			if result:
				return {
					"message": "created patient"
				}

			return {
				"error": "Could not create patient"
			}

File: criticare_app/lib/api/patient.dart
import 'package:http/http.dart' as http;
import '../models/patient.dart'; 
import 'dart:async';
import 'dart:convert';


class PatientApi {


    createPatient(Map<String, dynamic> formData) async {

        final _client = http.Client();
        var _loginUrl = Uri.parse('http://127.0.0.1:5000/patient');

        final response = await http.post(_loginUrl,
            headers: <String, String>{
                'Content-Type': 'application/json; charset=UTF-8',
            }, 
            body: jsonEncode({
                'ssn': formData['ssn'],
                'first_name': formData['firstName'], 
                'middle_name': formData['middleName'], 
                'last_name': formData['lastName'], 
                'date_of_birth': formData['dob'], 
                'height': formData['height'], 
                'weight': formData['weight'], 
                'next_of_kin': formData['nextOfKin'], 
                'home_phone': formData['homePhone'], 
                'work_phone': formData['workPhone'], 
                'symptoms': formData['symptoms'], 
                'health_insurance': formData['healthInsurance'], 
                'vaccination': formData['vaccination'], 
                'vaccination_date': formData['vaccinationDate'], 
                'drugs_alchohol': formData['drugsAlcohol'], 
                'sexually_active': formData['sexuallyActive'], 
                'allergies': formData['allergies'], 
                'blood_type': formData['bloodType'], 
                'notes': formData['notes']
            }),
        );
        //if this user gets a token, send them to the home page
        if (response.statusCode == 200) {
            print('Created Patient');
            return "Success"; 
        } else {
            return "Failure"; 
        }
    }

    static Future<List<Patient>> fetchPatients() async {
        final response = await http.get(Uri.parse('http://127.0.0.1:5000/patient'));
        if (response.statusCode == 200) {
            // If the server did return a 200 OK response,
            // then parse the JSON.
            final List<Patient> patients = patientFromJson(response.body);
            return patients;
        } else {
            // If the server did not return a 200 OK response,
            // then throw an exception.
            throw Exception('Failed to load album');
        }
    }

}